libs/ui/dialog/brain/src/lib/brn-dialog.service.ts

```diff
@@ -80,16 +80,21 @@ export class BrnDialogService implements OnDestroy {
 	private _backdrop: HTMLElement | null = null;
 
 	public closed = new EventEmitter<void>();
+	public stateChange = new EventEmitter<'open' | 'closed'>();
 
 	constructor() {
-		effect(() => {
-			if (this._overlay) {
-				this._renderer.setAttribute(this._overlay, 'data-state', this.state());
-			}
-			if (this._backdrop) {
-				this._renderer.setAttribute(this._backdrop, 'data-state', this.state());
-			}
-		});
+		effect(
```

Comment:
In theory, I have no issue with this change. I feel like I remember there being an infinite loop that caused me to remove the stateChanged directly. This might have been an early signal related issue though. Could you add an e2e test that makes sure your expected behavior works as intended? Thank you!

Reply:
Hi, of course😀, I will try to do it this weekend.